multiaddr_test.go

```diff
@@ -374,7 +382,7 @@ func TestFuzzBytes(t *testing.T) {
 }
 
 func randMaddrString() string {
-	good_corpus := []string{"tcp", "ip", "udp", "ipfs", "0.0.0.0", "127.0.0.1", "12345", "QmbHVEEepCi7rn7VL7Exxpd2Ci9NNB6ifvqwhsrbRMgQFP"}
+	good_corpus := []string{"tcp", "ip", "udp", "ipfs", "shs", "0.0.0.0", "127.0.0.1", "12345", "QmbHVEEepCi7rn7VL7Exxpd2Ci9NNB6ifvqwhsrbRMgQFP", ""}
```

Comment:
No, empty is not fine. Copypaste mistake.

I realized that the shs protocol only uses ed25519 keys, so I changed it from length prefixed var size to `ed25519.PublicKeySize` and added length checks in `codec.go`. The test still passes both with the correct and with empty string, though.

Should I import agl's ed25519 just for using that constant or should I copy it into this package? In case I should import it, how? By vendoring, gx or directly?